enterprise_access/apps/content_assignments/api.py

```diff
@@ -309,9 +380,17 @@ def _create_new_assignments(assignment_configuration, learner_emails, content_ke
             content_quantity=content_quantity,
             state=LearnerContentAssignmentStateChoices.ALLOCATED,
         )
-        assignment.full_clean()
         assignments_to_create.append(assignment)
 
+    # Next, try to populate the lms_user_id field on all assignments to be created (resulting in reads against User).
+    # Note: This covers the case where an admin assigns content to a learner AFTER they register.  For the case where an
+    # admin assigns content to a learner BEFORE they register, see the User post_save hook implemented in signals.py.
+    _try_populate_assignments_lms_user_id(assignments_to_create)
```

Comment:
nit: is it worth storing the result of this call in a variable, and then validating/bulk-creating below?  I know it doesn't functionally matter because `_try_populate_assignments_lms_user_id()` has the side-effect of possibly updating every assignment in `assignments_to_create`, but it's possibly confusing.

Reply:
Can you elaborate more?  My understanding is that a non-strict superset of the assignment objects with updated lms_user_ids are already being validated/bulk-created below.